core/src/test/scala/integration/kafka/api/AdminClientIntegrationTest.scala

```diff
@@ -811,6 +811,87 @@ class AdminClientIntegrationTest extends IntegrationTestHarness with Logging {
       assertEquals(3, servers(i).replicaManager.getReplica(topicPartition).get.logStartOffset)
   }
 
+  @Test
+  def testReplicaCanFetchFromLogStartOffsetAfterDeleteRecords(): Unit = {
+    val leaders = createTopic(topic, numPartitions = 1, replicationFactor = serverCount)
+    val followerIndex = if (leaders(0) != servers(0).config.brokerId) 0 else 1
+
+    def waitForFollowerLSOandLEOAfterRestart(expectedLSO: Long, expectedLEO: Long): Unit = {
```

Comment:
nit: maybe we can spell these out? Unfortunately "LSO" is also used for "last stable offset."